### ISSUE TITLE
Fails to build any package - - Windows 10

### ISSUE THREAD

**Issue by simonbjork** [microsoft/vcpkg#9394]
**Host Environment**
- OS: [Windows 10.]

**To Reproduce**
`vcpkg install vcpkg install freetype:x64-windows`

**Failure logs** 

Log in consolse:
```
D:\software_dev\vcpkg>vcpkg install freetype:x64-windows
The following packages will be built and installed:
  * bzip2[core]:x64-windows
    freetype[core]:x64-windows
  * libpng[core]:x64-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/4: zlib:x64-windows
Building package zlib[core]:x64-windows...
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Extracting source D:/software_dev/vcpkg/downloads/zlib1211.tar.gz
-- Applying patch cmake_dont_build_more_than_needed.patch
-- Using source at D:/software_dev/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb
-- Configuring x64-windows
CMake Error at software_dev/vcpkg/scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: ninja -v
    Working Directory: D:/software_dev/vcpkg/buildtrees/zlib/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\software_dev\vcpkg\buildtrees\zlib\config-x64-windows-out.log

Call Stack (most recent call first):
  software_dev/vcpkg/scripts/cmake/vcpkg_configure_cmake.cmake:293 (vcpkg_execute_required_process)
  software_dev/vcpkg/ports/zlib/portfile.cmake:22 (vcpkg_configure_cmake)
  software_dev/vcpkg/scripts/ports.cmake:94 (include)


Error: Building package zlib:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: zlib:x64-windows
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.
```

Here's the config-x64-windows-out.log from buildtrees/zlib.
```
[1/2] cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "D:/software_dev/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DSKIP_INSTALL_HEADERS=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/software_dev/vcpkg/packages/zlib_x64-windows/debug""
FAILED: ../../x64-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "D:/software_dev/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DSKIP_INSTALL_HEADERS=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/software_dev/vcpkg/packages/zlib_x64-windows/debug""
The system cannot find the path specified.
[2/2] cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "D:/software_dev/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/software_dev/vcpkg/packages/zlib_x64-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "D:/software_dev/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/software_dev/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/software_dev/vcpkg/packages/zlib_x64-windows""
CMake Warning at ../../../../scripts/buildsystems/vcpkg.cmake:104 (message):
  There are no libraries installed for the Vcpkg triplet x64-windows.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.15/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:4 (project)


-- The C compiler identification is MSVC 19.24.28314.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe -- broken
CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: D://CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_5dd79 && [1/2] Building C object CMakeFiles\cmTC_5dd79.dir\testCCompiler.c.obj
    [2/2] Linking C executable cmTC_5dd79.exe
    FAILED: cmTC_5dd79.exe 
    cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=CMakeFiles\cmTC_5dd79.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\link.exe /nologo CMakeFiles\cmTC_5dd79.dir\testCCompiler.c.obj  /out:cmTC_5dd79.exe /implib:cmTC_5dd79.lib /pdb:cmTC_5dd79.pdb /version:0.0  /machine:x64  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:4 (project)


-- Configuring incomplete, errors occurred!
See also "D://CMakeFiles/CMakeOutput.log".
See also "D://CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.
```

**Additional context**

I've tried reinstalling Visual Studio (Community), reinstalling vcpkg. Tried in multiple computers.

- Download and install Visual Studio Community
- Clone vcpkg and install with bootstrap-vcpkg.bat
- run any buld command, it all errors.

I've tried with a custom installed cmake and also having vcpkg install the portable version.

**Comment by plettplett** [comment]
Hi Simon,

This is chris@digitaldevoid.com. I see you're likely in the right place to try get openimageio built on windows 10. I've been running into snags, and have had to work my way through them myself. Working backwards, I've uninstalled VS 2019, and instead using VS 2017 (update 15.9). Just to be safe, all the openimageio dependency & optional libs were removed through vcpkg, then reinstalled through vcpkg using VS2017's compiler. But in doing so I had 2 concurrent vcpkg instances running, which (I assume) corrupted the vcpkg database. I reinstalled vcpkg, but of course the db is still corrupted! Now I need to find the actual db to manually remove it. 

Prior to this, there were a couple of things that worked for me, but for some very odd reason the "vcpkg install  openimageio" seems to complete the install, but says "Build failed" at the end of the script. Indeed, everything is there (in various directories) except the libs!! Even the log file checks out perfectly. Not sure what to make of it. But see this for help: https://github.com/microsoft/vcpkg/issues/8812. I ended up with the last option, by editing ${vcpkg_root}/triplets/x64-windows.cmake by placing this in the first line:

 if(${PORT} MATCHES "openimageio")
    set(VCPKG_PLATFORM_TOOLSET v141)
endif()
